src/redprl/operator.sml

```diff
@@ -88,10 +88,86 @@ structure RedPrlParameterTerm = AbtParameterTerm (RedPrlParameter)
 
 structure RedPrlArity = ListAbtArity (structure PS = RedPrlParamSort and S = RedPrlSort)
 
+
+structure RedPrlKind =
+struct
+  (*
+   * DISCRETE < KAN < HCOM < PRE
+   *                < COE  <
+   *
+   * and KAN = meet (HCOM, COE)
+   *)
+  datatype kind = DISCRETE | KAN | HCOM | COE | PRE
+
+  val toString =
+    fn DISCRETE => "discrete"
+     | KAN => "kan"
+     | HCOM => "hcom"
+     | COE => "coe"
+     | PRE => "pre"
+
+  local
+    structure Internal :
+    (* this could be the new meet semi-lattice *)
+    sig
+      type t = kind
+      val top : t
+      val leq : t * t -> bool
+      val meet : t * t -> t
```

Comment:
@favonia Yeah; in the future, one option that I favor would be to define new signatures inside the RedPRL project until they are perfected, and after that, move them into whatever library makes sense. that can lower the cost.

But in this case, do whatever you feel is most convenient for you.